### PR TITLE
adding kms wallet

### PR DESCRIPTION
**Pull request by ibadia**
* Adding KMS support for posting orders
- User creates an KMSSIGNER object and passes it to the fireflyclient and for private key user can just pass '' <emptystring>.
- library will be initialised using kms
- user will be onboarded using kms
- user posting order will be done through kms.

https://linear.app/seed/issue/QUANT-552/add-support-for-kms-in-firefly-library

### CODE REVIEW

File: examples/23-postOrderkms.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/**
```
Comment (YameenMalik) [inline examples/23-postOrderkms.ts:1]: capitalize `KMS` in file name "23-postOrderUsingKMS.ts"

File: src/fireflyClient.ts [hunk 2]
```diff
@@ -174,6 +184,22 @@ export class FireflyClient {
     if (_acctPvtKey) {
       this.initializeWithPrivateKey(_acctPvtKey);
     }
+    if (_awsKmsSigner){
+      this.initializeWithKMS(_awsKmsSigner);
+    }
+
+  }
+
+  initializeWithKMS=async (awsKmsSigner: AwsKmsSigner): Promise<void> => {
```
Comment (YameenMalik) [inline src/fireflyClient.ts:193]: Add space before and after `=`

**Review by YameenMalik — CHANGES_REQUESTED** [review]
Please look at the comments, before we can go on and merge it to main.

File: src/fireflyClient.ts [hunk 3]
```diff
@@ -174,6 +184,22 @@ export class FireflyClient {
...
+    try {
+      this.kmsSigner=awsKmsSigner;
+      const address=await this.kmsSigner.getAddress();
```
[the remark this answers is not in the document]
Comment (ibadia) [reply src/fireflyClient.ts:196]: done

File: examples/23-postOrderkms.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+async function main() {
+
+  const kmsKeyArn='arn:aws:kms:ap-northeast-1:648912772077:key/254a5c49-9a68-48ca-81f7-8715de8dca73'
+  const kmsSigner= new AwsKmsSigner({region: 'ap-northeast-1', keyId: kmsKeyArn})
+
+
+
+  const dummyAccountKey='';
+  const client = new FireflyClient(true, Networks.TESTNET_ARBITRUM, dummyAccountKey,kmsSigner); //passing isTermAccepted = true for compliance and authorizarion
```
[the remark this answers is not in the document]
Comment (ibadia) [reply examples/23-postOrderkms.ts]: acctPvtKey is not required anymore as now it accepts either privatekey or kms object

File: src/fireflyClient.ts [hunk 5]
```diff
@@ -151,11 +159,13 @@ export class FireflyClient {
    * @param _isTermAccepted boolean indicating if exchange terms and conditions are accepted
    * @param _network containing network rpc url and chain id
    * @param _acctPvtKey private key for the account to be used for placing orders
+   * @param _awsKmsSigner kms signer object for signing transactions via kms
    */
   constructor(
     _isTermAccepted: boolean,
     _network: ExtendedNetwork,
-    _acctPvtKey?: string
+    _acctPvtKey?: string,
```
[the remark this answers is not in the document]
Comment (ibadia) [reply src/fireflyClient.ts]: yes, have changed it.

File: src/fireflyClient.ts [hunk 6]
```diff
@@ -174,6 +184,22 @@ export class FireflyClient {
     if (_acctPvtKey) {
       this.initializeWithPrivateKey(_acctPvtKey);
     }
+    if (_awsKmsSigner){
```
[the remark this answers is not in the document]
Comment (ibadia) [reply src/fireflyClient.ts]: added an else if now

**Review by YameenMalik — APPROVED** [review]
Looks good.

**Pull request merged**